ecommerce/frontend/src/components/profile/ProfileDetails.js

```diff
@@ -1,10 +1,24 @@
 import './ProfileDetails.less'
-import { Link } from 'react-router-dom'
-import { Tabs } from 'antd'
+import { Link, Redirect } from 'react-router-dom'
+import { useState } from 'react'
+import { Button, Tabs } from 'antd'
+import { api } from '../../api'
+import { useAppContext } from '../../context/AppContext'
 
 const { TabPane } = Tabs
 
 export const ProfileDetails = () => {
+    const [isLoading, setIsLoading] = useState(false)
+    const { user } = useAppContext()
+
+    const onVerify = () => {
+        async function verify() {
+            setIsLoading(true)
+            const { data } = await api.post('/user/verify', {})
+            setIsLoading(false)
+        }
+        verify()
+    }
```

Comment:
for a callback you don't need to create an inner async function you can simply write
```js
const onVerify = async () => {
   setIsLoading(true)
   ...
}
```

Reply:
also make sure you add try/catch/finally because if that api call fails for some reason then setIsLoading(false) would never happen leaving user with never ending loading spinner

Reply:
you would have the setIsLoading(false) in the finally scope. Also show an error notification